pacman/executor/pacman_algorithm_executor.py

```diff
@@ -391,27 +391,26 @@ def _determine_algorithm_order(
             else:
 
                 # Failed to find an algorithm to run!
-                algorithms_left_names = list()
+                algorithms_to_find_names = list()
                 for algorithm in algorithms_to_find:
-                    algorithms_left_names.append(algorithm.algorithm_id)
+                    algorithms_to_find_names.append(algorithm.algorithm_id)
+                optional_algorithms_names = list()
                 for algorithm in optional_algorithm_data:
-                    algorithms_left_names.append(algorithm.algorithm_id)
+                    optional_algorithms_names.append(algorithm.algorithm_id)
                 algorithms_used = list()
                 for algorithm in allocated_algorithms:
                     algorithms_used.append(algorithm.algorithm_id)
                 algorithm_input_requirement_breakdown = ""
                 for algorithm in algorithms_to_find:
-                    if algorithm.algorithm_id in algorithms_left_names:
-                        algorithm_input_requirement_breakdown += \
-                            self._deduce_inputs_required_to_run(
-                                algorithm, input_types, token_states,
-                                fake_inputs, fake_tokens)
+                    algorithm_input_requirement_breakdown += \
+                        self._deduce_inputs_required_to_run(
+                            algorithm, input_types, token_states,
+                            fake_inputs, fake_tokens)
                 for algorithm in optionals_to_use:
-                    if algorithm.algorithm_id in algorithms_left_names:
```

Comment:
by removing this and the one above, i THINK (but to be honest, id need to run tests to verify) that you've opened up a risk of running the same algorithm multiple times. But it might be picked up later on due to not producing any new outputs.......but not sure.

Reply:
By this point, an error is about to be produced I think...

Reply:
having opened the entire file, yep ya right. 
nail in why i hate using githubs interface. lol. i dont see it in all context. 

on that note, happy to merge